### PR TITLE
TRUS-3598 Re-write connector tests to not pass the HeaderCarrier

### PR DESCRIPTION
**Pull request by carrielies**

### CODE REVIEW

File: app/connectors/TrustDataConnector.scala [hunk 1]
```diff
@@ -40,15 +41,15 @@ class TrustDataConnector @Inject()(http: HttpClient, config: AppConfig) extends
...
-  def getTrustJson(identifier: Identifier)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[TrustDataResponse] = {
+  def getTrustJson(identifier: Identifier): Future[TrustDataResponse] = {
     lazy val url: String = s"${config.trustDataUrl}/trusts/obliged-entities/$identifier/${identifier.value}"
 
     val correlationId = UUID.randomUUID().toString
 
-    val headersWithoutOldAuth = hc.copy(authorization = None)
-    val hcExtra: HeaderCarrier = headersWithoutOldAuth.withExtraHeaders(trustDataHeaders(correlationId): _*)
+    implicit val hc: HeaderCarrier = HeaderCarrier(authorization = None, extraHeaders = trustDataHeaders(correlationId))
     logger.info(s"[Session ID: ${Session.id(hc)}] getTrustJson correlationId: $correlationId from call to url: $url")
```
Comment (jackheslop96) [inline app/connectors/TrustDataConnector.scala:50]: Is `Session.id(hc)` now always going to return "No Session ID available" ?

File: app/connectors/TrustDataConnector.scala [hunk 2]
```diff
@@ -40,15 +41,15 @@ class TrustDataConnector @Inject()(http: HttpClient, config: AppConfig) extends
...
-    http.GET[TrustDataResponse](url)(TrustDataResponse.httpReads(identifier), hcExtra, ec)
+
+    http.GET[TrustDataResponse](url)(TrustDataResponse.httpReads(identifier), implicitly[HeaderCarrier](hc), global)
```
Comment (jackheslop96) [inline app/connectors/TrustDataConnector.scala:52]: Out of interest, how does `implicitly[HeaderCarrier](hc)` differ from just passing in `hc`?

File: app/connectors/TrustDataConnector.scala (hunk 1 above)
Comment (carrielies) [reply app/connectors/TrustDataConnector.scala:50]: Hmmm. It looks like it probably will, but I basically just copied it from Trusts. For my logging, it's the CorrelationId I really care about, so I don't mind

File: app/connectors/TrustDataConnector.scala (hunk 2 above)
Comment (carrielies) [reply app/connectors/TrustDataConnector.scala:52]: I don't know. I just copied exactly what Trusts is doing. I had the same thought that it probably would do nothing, but I just want the call to work so thought it was best to copy it exactly

Comment (jackheslop96) [reply app/connectors/TrustDataConnector.scala:52]: Fair enough

**Pull request merged**